resources/prosody-plugins/mod_filter_iq_jibri.lua

```diff
@@ -10,12 +15,17 @@ module:hook("pre-iq/full", function(event)
         if jibri then
             local session = event.origin;
             local token = session.auth_token;
+            local room = get_room_from_jid(room_jid_match_rewrite(jid_bare(stanza.attr.to)));
+            local occupant = room:get_occupant_by_real_jid(stanza.attr.from);
+            local feature = jibri.attr.recording_mode == 'file' and 'recording' or 'livestreaming';
+            local is_allowed = is_feature_allowed(
+                feature,
+                session.jitsi_meet_context_features,
+                session.granted_jitsi_meet_context_features,
+                occupant.role == 'moderator');
 
             if jibri.attr.action == 'start' then
-                if token == nil
-                    or not is_feature_allowed(session.jitsi_meet_context_features,
-                    (jibri.attr.recording_mode == 'file' and 'recording' or 'livestreaming')
-                ) then
+                if token == nil or not is_allowed then
```

Comment:
Do we still need to check `token` here? What are we supposed to do if the user is a moderator with no token? Or if the user was granted features, but has no token?

If some form of check for `token` is necessary can we move it to `is_feature_allowed`?

Reply:
> What are we supposed to do if the user is a moderator with no token?

The case when you are not using jwt for authentication, but you are moderator. Hum but this is explicit check if there is no token ... I will drop it.



> Or if the user was granted features, but has no token?

Most of the cases are like this, you grant moderation to guests that has no token.

Reply:
Done